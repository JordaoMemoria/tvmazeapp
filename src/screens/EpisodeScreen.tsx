import React, {useEffect, useState, useLayoutEffect} from 'react';
import styled from 'styled-components/native';
import {NavigationProps} from '../typescript/interfaces';
import Loading from '../components/Loading';
import Item from '../components/Item';
import {getEpisode} from '../tvmaze/api';
import {EpisodeDetail} from '../typescript/interfaces';
import {formatSummary} from '../common/utils';
import WhiteSquare from '../components/WhiteSquare';
import BlueSquare from '../components/BlueSquare';
import SmallHeight from '../components/SmallHeight';

export default function EpisodeScreen({route, navigation}: NavigationProps) {
  const {id, season, number, name, serieName} = route.params;
  const [episode, setEpisode] = useState<EpisodeDetail>() as any;

  useLayoutEffect(() => {
    navigation.setOptions({
      title: serieName,
    });
  }, [navigation]);

  useEffect(() => {
    getEpisode(id, (data: EpisodeDetail) => {
      setEpisode(data);
    });
  }, []);

  return (
    <Scroll>
      {episode ? (
        <>
          <BlueSquare text={`Season ${season}`} center />
          <Item
            id={id}
            canClick={false}
            name={`${number}. ${name}`}
            image={episode.image !== null ? episode.image.medium : ''}
            category="Episode"
            favorite={false}
            showFavorite={false}
          />
          <WhiteSquare text={formatSummary(episode.summary)} />
          <SmallHeight />
        </>
      ) : (
        <Loading margin={50} />
      )}
    </Scroll>
  );
}

const Scroll = styled.ScrollView``;
