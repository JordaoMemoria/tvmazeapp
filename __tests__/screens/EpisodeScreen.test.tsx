import React from 'react'
import EpisodeScreen from '../../src/screens/EpisodeScreen'
import renderer from 'react-test-renderer'

jest.useFakeTimers()

describe('<EpisodeScreen />', () => {
  it('Renders correctly', () => {
    const mockedParams = {
      route: {
        params: {
          id: 2,
          season: 1,
          number: 5,
          name: 'Rickverso',
          serieName: 'Rick and Morty',
        },
      },
      navigation: {
        setOptions: (props: { title: 'Rick and Morty' }) => {},
      },
    }

    const tree = renderer.create(<EpisodeScreen {...mockedParams} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
