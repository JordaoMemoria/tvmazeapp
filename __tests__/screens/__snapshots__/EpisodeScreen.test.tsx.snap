// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EpisodeScreen /> Renders correctly 1`] = `
<RCTScrollView
  style={
    Array [
      Object {},
    ]
  }
>
  <View>
    <ActivityIndicator
      color="#5AB5E8"
      margin={50}
      size="large"
      style={
        Array [
          Object {
            "marginTop": 50,
          },
        ]
      }
    />
  </View>
</RCTScrollView>
`;
